src/System.Text.Json/src/System/Text/Json/JsonException.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Defines a custom exception object that is thrown when invalid JSON text is encountered, when the defined maximum depth is passed,
```

Comment:
nit: Is this capturing all the cases where `JsonException` can be thrown? Should it? If we throw `JsonException` on a edge case, we would have to update the comment. Maybe we should keep it more generic and not enumerate all the cases.

Reply:
Note that `JsonException` is abstract currently so it can't be thrown directly. I'll re-word to avoid listing cases.

Reply:
We should update the comment here to match the current semantics but don't list all the "error" cases in the summary (maybe as remarks).